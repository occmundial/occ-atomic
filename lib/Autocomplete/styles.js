import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
