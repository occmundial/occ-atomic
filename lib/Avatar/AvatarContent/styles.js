import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
