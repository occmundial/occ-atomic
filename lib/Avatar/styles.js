import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
