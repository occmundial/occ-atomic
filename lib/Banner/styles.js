import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
