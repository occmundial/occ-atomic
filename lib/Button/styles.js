import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
