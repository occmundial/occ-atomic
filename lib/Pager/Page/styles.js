import colors from '../../tokens/colors.json';
import fonts from '../../tokens/fonts.json';
import spacing from '../../tokens/spacing.json';
import borderRadius from '../../tokens/borderRadius.json';
import shadows from '../../tokens/shadows.json';

import legacyFonts from '../../subatomic/fonts';

const objectToFontValue = (font) =>
    `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFonts.body}`;

export default {
    page: {
        display: 'flex',
        cursor: 'pointer',
        minWidth: 45,
        padding: [spacing['size-3'], spacing['size-2']],
        justifyContent: 'center',
        alignItems: 'center',
        gap: spacing['size-2'],
        borderRadius: borderRadius['br-xs'],
        background: 'transparent',
        font: objectToFontValue(fonts['button-small']),
        color: colors.text.corp.secondary,
        transition: '0.3s all',
        '&:focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        outline: 'none',
        border: 'none',
        '&:hover': {
            background: colors.bg.action.ghost.hover
        },
        '&:active': {
            background: colors.bg.action.ghost.active
        }
      },
    static: {
        pointerEvents: 'none'
    },
    active: {
        background: `${colors.bg.action.brand.default} !important`,
        color: `${colors.text.white.primary} !important`,
        zIndex: 1
    },
    disabled: {
        pointerEvents: 'none',
        background: 'transparent',
        color: colors.text.corp.disabled
    }
};