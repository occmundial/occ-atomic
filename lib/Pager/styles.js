import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';

export default {
    pager: {
        listStyle:'none',
        paddingLeft:'0',
        fontFamily:fonts.body,
        color:colors.grey7,
        fontSize:'14px'
    },
    btn: {
        display:'inline-block',
        fontFamily:fonts.body,
        border:`1px solid ${colors.grey1}`,
        borderRadius:'5px',
        padding:'3px 13px',
        transition:'0.3s all',
        cursor:'pointer',
        outline:'0',
        '& span': {
            marginBottom:'-2px'
        },
        '&:hover': {
            background:'#ECECEE',
            borderColor:'#ECECEE'
        }
    },
    prev: {
        marginRight:'15px',
        '& span': {
            transform:'rotate(90deg)'
        }
    },
    next: {
        marginLeft:'15px',
        '& span': {
            transform:'rotate(-90deg)'
        }
    },
    disabled: {
        pointerEvents:'none',
        opacity:'0.4'
    }
};