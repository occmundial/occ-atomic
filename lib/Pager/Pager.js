import React, { useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';

import Button from '../Button';
import Page from './Page';

/** This pager allows the user to navigate between pages with 'Next' and 'Previous' buttons and numerical buttons. */
export default function Pager({
  classes,
  currentPage,
  onPageChange,
  pageCount = 10,
  centerPages = 3,
  marginPages = 2,
  previousLabel = 'Previous',
  nextLabel = 'Next',
  breakSymbol = '...',
  hideNumbers,
  disabled,
  className
}) {
  const handlePageSelected = useCallback(
    (newPage) => {
      if (currentPage === newPage) return;
      if (onPageChange) onPageChange(newPage);
    },
    [currentPage, onPageChange]
  );

  useEffect(() => {
    if (currentPage > pageCount || currentPage < 1) {
      handlePageSelected(1);
    }
  }, [currentPage, pageCount, handlePageSelected]);

  const handlePrevPage = () => {
    if (currentPage > 1) handlePageSelected(currentPage - 1);
  };

  const handleNextPage = () => {
    if (currentPage < pageCount) handlePageSelected(currentPage + 1);
  };

  const addPage = (page) => {
    const pageData = {
      key: page.toString(),
      type: 'page',
      selected: currentPage === page
    };
    return pageData;
  };

  const getStartAndEndPages = () => {
    let startPage = Math.max(
      currentPage - Math.floor(centerPages / 2),
      marginPages + 1
    );
    let endPage = Math.min(startPage + centerPages - 1, pageCount - marginPages);

    if (endPage === pageCount - marginPages) {
      startPage = Math.max(endPage - centerPages + 1, marginPages + 1);
    }

    return { startPage, endPage };
  };

  function getPagination() {
    const elements = [];

    if (pageCount <= centerPages + marginPages * 2) {
      for (let i = 1; i <= pageCount; i++) {
        elements.push(addPage(i));
      }
    } else {
      for (let i = 1; i <= marginPages; i++) {
        elements.push(addPage(i));
      }

      const { startPage, endPage } = getStartAndEndPages();

      if (currentPage > marginPages + 1 && startPage > marginPages + 1) {
        elements.push({ key: 'left-break', type: 'break' });
      }

      for (let i = startPage; i <= endPage; i++) {
        elements.push(addPage(i));
      }

      if (currentPage < pageCount - marginPages - 1) {
        elements.push({ key: 'right-break', type: 'break' });
      }

      for (let i = pageCount - marginPages + 1; i <= pageCount; i++) {
        elements.push(addPage(i));
      }
    }

    return elements;
  }

  return (
    <div className={`${classes.pager}${className ? ` ${className}` : ''}`}>
      <Button
        className={classes.prev}
        disabled={currentPage === 1 || disabled}
        theme="secondary"
        onClick={handlePrevPage}
        icon="arrow-left"
        size={!previousLabel ? 'lg' : 'sm'}
      >
        {previousLabel}
      </Button>
      {!hideNumbers &&
        getPagination().map(({ key, type, selected }) => {
          if (type === 'page') {
            return (
              <Page
                key={key}
                onClick={() => handlePageSelected(Number(key))}
                selected={selected}
                page={Number(key)}
                disabled={disabled}
              />
            );
          }
          return <Page key={key} page={breakSymbol} disabled={disabled} />;
        })
      }
      <Button
        className={classes.next}
        disabled={currentPage === pageCount || disabled}
        theme="secondary"
        onClick={handleNextPage}
        iconRight="arrow-right"
        size={!nextLabel ? 'lg' : 'sm'}
      >
        {nextLabel}
      </Button>
    </div>
  );
}

Pager.propTypes = {
  classes: PropTypes.object,
  /** The current page number. */
  currentPage: PropTypes.number.isRequired,
  /** The total number of pages. */
  pageCount: PropTypes.number,
  /** The number of page numbers displayed at the center, between the '...' separators. */
  centerPages: PropTypes.number,
  /** The number of page numbers displayed on the sides of the pager. */
  marginPages: PropTypes.number,
  /** The label for the 'Previous' button. */
  previousLabel: PropTypes.node,
  /** The label for the 'Next' button. */
  nextLabel: PropTypes.node,
  /** Function to call after clicking a button to change the page. */
  onPageChange: PropTypes.func,
  /** Hide the numerical buttons, leaving just the 'Next' and 'Previous' buttons to interact with. */
  hideNumbers: PropTypes.bool,
  /** Label for the separators. */
  breakSymbol: PropTypes.string,
  /** Disable the pager */
  disabled: PropTypes.bool,
  className: PropTypes.string
};
