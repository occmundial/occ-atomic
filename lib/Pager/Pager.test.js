import React from 'react';
import { shallow, mount } from 'enzyme';
import Pager from './Pager';
import styles from './styles';
import PagerJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Pager", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Pager classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the prev and next buttons', () => {
        const wrapper = shallow(
            <Pager classes={classes} />
        );
        expect(wrapper.find('.prev').length).toBe(1);
        expect(wrapper.find('.prev').text()).toEqual('<Icon /> Previous');
        expect(wrapper.find('.next').length).toBe(1);
        expect(wrapper.find('.next').text()).toEqual('Next <Icon />');
    });

    it('changes to previous page', () => {
        const wrapper = shallow(
            <Pager initialPage={3} classes={classes} />
        );
        expect(wrapper.state('selected')).toEqual(3);
        wrapper.find('.prev').simulate('click', {target: {}});
        expect(wrapper.state('selected')).toEqual(2);
    });

    it('changes to next page', () => {
        const wrapper = shallow(
            <Pager initialPage={3} classes={classes} />
        );
        expect(wrapper.state('selected')).toEqual(3);
        wrapper.find('.next').simulate('click', {target: {}});
        expect(wrapper.state('selected')).toEqual(4);
    });

    it('renders the pagination', () => {
        const wrapper = mount(
            <Pager pageCount={3} pageRangeDisplayed={1} marginPagesDisplayed={1} classes={classes} />
        );
        expect(wrapper.find('li').length).toBe(5);
    });

    it('changes to a specific page', () => {
        const wrapper = mount(
            <Pager pageCount={5} pageRangeDisplayed={1} marginPagesDisplayed={1} classes={classes} />
        );
        expect(wrapper.state('selected')).toEqual(1);
        wrapper.find('li').at(3).simulate('click', {target: {}});
        expect(wrapper.state('selected')).toEqual(5);
    });

    it('disables the previous page button', () => {
        const wrapper = shallow(
            <Pager initialPage={1} classes={classes} />
        );
        expect(wrapper.find('.prev').hasClass('disabled')).toEqual(true);
        wrapper.setProps({forcePage:5});
        expect(wrapper.find('.prev').hasClass('disabled')).toEqual(false);
    });

    it('disables the next page button', () => {
        const wrapper = shallow(
            <Pager pageCount={5} initialPage={5} classes={classes} />
        );
        expect(wrapper.find('.next').hasClass('disabled')).toEqual(true);
        wrapper.setProps({forcePage:1});
        expect(wrapper.find('.next').hasClass('disabled')).toEqual(false);
    });

    it('calls the onPageChange function', () => {
        const onPageChange = jest.fn();
        const wrapper = shallow(
            <Pager onPageChange={onPageChange} classes={classes} />
        );
        wrapper.find('.next').simulate('click', {target: {}});
        expect(onPageChange.mock.calls.length).toBe(1);
    });

});

describe("PagerJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <PagerJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Pager styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});
