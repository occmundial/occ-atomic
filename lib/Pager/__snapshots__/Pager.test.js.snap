// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`Pager styles matches the snapshot 1`] = `
Object {
  "btn": Object {
    "& span": Object {
      "marginBottom": "-2px",
    },
    "&:hover": Object {
      "background": "#ECECEE",
      "borderColor": "#ECECEE",
    },
    "border": "1px solid #bfbfbf",
    "borderRadius": "5px",
    "cursor": "pointer",
    "display": "inline-block",
    "fontFamily": "'HK Grotesk', sans-serif",
    "outline": "0",
    "padding": "3px 13px",
    "transition": "0.3s all",
  },
  "disabled": Object {
    "opacity": "0.4",
    "pointerEvents": "none",
  },
  "next": Object {
    "& span": Object {
      "transform": "rotate(-90deg)",
    },
    "marginLeft": "15px",
  },
  "pager": Object {
    "color": "#727272",
    "fontFamily": "'HK Grotesk', sans-serif",
    "fontSize": "13px",
    "listStyle": "none",
    "paddingLeft": "0",
  },
  "prev": Object {
    "& span": Object {
      "transform": "rotate(90deg)",
    },
    "marginRight": "15px",
  },
}
`;

exports[`PagerJSS matches the snapshot 1`] = `ShallowWrapper {}`;
